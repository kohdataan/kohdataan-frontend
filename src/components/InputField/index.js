import React, { memo } from 'react'
import PropTypes from 'prop-types'

const InputField = props => {
  const {
    label,
    showLabel,
    value,
    inputClassName,
    labelClassName,
    onChange,
    type,
  } = props
  return (
    <label htmlFor={label} className={inputClassName}>
      {showLabel && label}
      <input
        type={type}
        name={label}
        id={label}
        aria-label={label}
        value={value}
        placeholder={label}
        className={labelClassName}
        onChange={onChange}
      />
    </label>
  )
}

InputField.propTypes = {
  type: PropTypes.string,
  label: PropTypes.string.isRequired,
  showLabel: PropTypes.bool,
  value: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  onChange: PropTypes.func.isRequired,
}

InputField.defaultProps = {
  type: 'text',
  showLabel: true,
  value: '',
  inputClassName: '',
  labelClassName: '',
}

export default memo(InputField)
