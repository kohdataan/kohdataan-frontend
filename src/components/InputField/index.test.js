import React from 'react'
import { create } from 'react-test-renderer'
import InputField from './index'

describe('Input field', () => {
  test('it matches the snapshot', () => {
    const component = create(<InputField label="test" onChange={() => {}} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
