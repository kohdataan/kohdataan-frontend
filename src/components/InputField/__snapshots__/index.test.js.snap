// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field it matches the snapshot 1`] = `
<label
  className=""
  htmlFor="test"
>
  test
  <input
    aria-label="test"
    aria-required={false}
    className=""
    id="test"
    name="test"
    onChange={[Function]}
    placeholder="test"
    type="text"
    value=""
  />
</label>
`;
